Update: use control-plane instead of master for node selector

zewolfe opened the pull request:
Towards - https://github.com/giantswarm/roadmap/issues/2135

webwurst commented on helm/coredns-app/templates/deployment-masters.yaml:
```diff
@@ -55,8 +55,13 @@ spec:
         {{-  range $key, $value := $.Values.mastersInstance.nodeSelector }}
-        {{ $key }}: {{ $value }}
-        {{- end}}
+          {{- if and (eq $key "node-role.kubernetes.io/control-plane") (lt (int $.Capabilities.KubeVersion.Minor) 25) }}
```
Maybe we can use `(semver "1.2.3").Compare` here? https://docs.helm.sh/docs/chart_template_guide/function_list/#semver

webwurst replied:
```suggestion
          {{- if and (eq $key "node-role.kubernetes.io/control-plane") (semverCompare "<1.25" .Capabilities.KubeVersion.Version) }}
```

webwurst commented:
I see the linter complaining, not sure why. I took the `semverCompare` from [ingress-nginx/templates/controller-service.yaml](https://github.com/kubernetes/ingress-nginx/blob/main/charts/ingress-nginx/templates/controller-service.yaml#L69):
```
{{- if and (semverCompare ">=1.20" .Capabilities.KubeVersion.Version) (.Values.controller.service.appProtocol) }}
```

zewolfe commented:
I think it was the scope, because `.Capabilities` isn't available in the block. I've updated it to reference from `$.Capabilities` instead. I've updated it and it looks okay.
